### PR TITLE
Recognize "manifest unknown" errors reported by Harbor

### PR DESCRIPTION
**Pull request by mtrmac**
... per data in https://github.com/containers/image/issues/2203 .

### CODE REVIEW

File: docker/docker_client.go [hunk 1]
```diff
@@ -1097,6 +1097,11 @@ func isManifestUnknownError(err error) bool {
 	if errors.As(err, &e) && e.ErrorCode() == errcode.ErrorCodeUnknown && e.Message == "Not Found" {
 		return true
 	}
+	// Harbor v2.10.2
+	if errors.As(err, &e) && e.ErrorCode() == errcode.ErrorCodeUnknown && strings.Contains(e.Message, "not found") {
```
Comment (rhatdan) [inline docker/docker_client.go:1101]: How about replace the error check above with:

```
if errors.As(err, &e) && e.ErrorCode() == errcode.ErrorCodeUnknown && strings.Contains(strings.ToLower(e.Message), "not found") {
```

Comment (mtrmac) [reply docker/docker_client.go]: Thanks, good idea. Updated.

**Comment by rhatdan** [comment]
LGTM
@giuseppe @vrothberg @mheon PTAL

**Review by vrothberg — APPROVED** [review]
LGTM
Merge at will ✅

**Comment by TomSweeneyRedHat** [comment]
LGTM
but it looks like the branch may need to be updated @mtrmac

**Comment by mtrmac** [comment]
For the record, this was confirmed to work in https://github.com/containers/image/issues/2203#issuecomment-2113978491 .

**Pull request merged**